### PR TITLE
Add <DynamicHome /> (behind a release flag)

### LINKED ISSUE #1755: Replace LandingContainer with DynamicHome
## Issue - Enhancement

This story effectively removes the landing page, and replaces it with logic which redirects you to either the authenticated home page (the user dashboard) or the new signin page based on your auth state.

### Acceptance Criteria (optional)
- When a user is unauthenticated, and hits the `/` route, they get redirected to `/signin`
- When a user *is* authenticated and hits the `/` route, they get redirected to `/me`

- **Estimated effort:** X points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

### PR DESCRIPTION
**Pull request by dan-f**
Fixes #1755.

This is one of the first steps for the new signin flow. It's hidden behind a release flag.

### CODE REVIEW

**Comment by dan-f** [comment]
Testing:

Do the following when you're already signed in, and when you're signed out:
  - [ ] Flip the `newSignIn` release flag to `true`
  - [ ] Run `yarn dev`
  - [ ] When you were previously signed in, and you load `http://localhost:3000/`, you should get redirected to the Team Dashboard
  - [ ] When you were previously signed out, and you load `http://localhost:3000/`, you should get redirected to `/signin`, which is not yet implemented ;)
  - [ ] Don't forget to reset/checkout the `releaseFlags.js` module

**Comment by jordanh** [comment]
Looking at this one now

File: src/universal/components/Action/Action.js [hunk 1]
```diff
@@ -26,7 +28,13 @@ const Action = (props) => {
       <Toast />
       <SocketHealthMonitor />
       <Switch>
-        <Route exact path="/" component={LandingContainer} />
+        {__RELEASE_FLAGS__.newSignIn
```
Comment (jordanh) [inline src/universal/components/Action/Action.js]: Minor, minor quibble; should this be called `__RELEASE_FLAGS__` (plural) or `__RELEASE_FLAG_` (singular)? I think the latter, right? It's a single boolean value, right?

**Comment by jordanh** [comment]
# Test Cycle 1
## Status: in progress

Do the following when you're already signed in, and when you're signed out:
  - [ ] Flip the `newSignIn` release flag to `true`
  - [ ] Run `yarn dev`
  - [ ] When you were previously signed in, and you load `http://localhost:3000/`, you should get redirected to the Team Dashboard
  - [ ] When you were previously signed out, and you load `http://localhost:3000/`, you should get redirected to `/signin`, which is not yet implemented ;)
  - [ ] Don't forget to reset/checkout the `releaseFlags.js` module

File: src/universal/components/Action/Action.js (hunk 1 above)
Comment (jordanh) [reply src/universal/components/Action/Action.js]: Oh crap, I'm an idiot. Nevermind. I see that this is actually an object exposed in `releaseFlags.js`!

**Comment by jordanh** [comment]
LGTM!

**Pull request merged**